uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/SicknessRequest.php

```diff
@@ -2,47 +2,50 @@
 
 use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
 use CRM_HRLeaveAndAbsences_BAO_SicknessRequest as SicknessRequest;
-use CRM_HRLeaveAndAbsences_Service_LeaveBalanceChange as LeaveBalanceChangeService;
 use CRM_HRLeaveAndAbsences_Service_LeaveRequest as LeaveRequestService;
 
-class CRM_HRLeaveAndAbsences_Service_SicknessRequest {
+class CRM_HRLeaveAndAbsences_Service_SicknessRequest extends CRM_HRLeaveAndAbsences_Service_LeaveRequest {
 
   /**
-   * @var \LeaveBalanceChangeService
+   * {@inheritDoc}
    */
-  private $leaveBalanceChangeService;
-
-  /**
-   * @var \LeaveRequestService
-   */
-  private $leaveRequestService;
-
-  public function __construct(
-    LeaveBalanceChangeService $leaveBalanceChangeService,
-    LeaveRequestService $leaveRequestService
-  ) {
-    $this->leaveBalanceChangeService = $leaveBalanceChangeService;
-    $this->leaveRequestService = $leaveRequestService;
+  protected function canChangeDatesFor($params) {
+    return true;
   }
 
   /**
-   * Wraps the SicknessRequest BAO create method in order to perform some more
-   * actions before/after calling it.
+   * This method creates/updates the SicknessRequest along with the
+   * associated LeaveRequest and it's LeaveBalanceChanges
    *
    * @param array $params
-   * @param bool $validate
    *
-   * @return \CRM_HRLeaveAndAbsences_BAO_SicknessRequest
+   * @return \CRM_HRLeaveAndAbsences_BAO_SicknessRequest|NULL
    */
-  public function create($params, $validate = true) {
-    $sicknessRequest = SicknessRequest::create($params, $validate);
+  protected function createLeaveRequestWithBalanceChange($params) {
```

Comment:
maybe we should rename this method to `createWithBalanceChanges` or `createRequestWithBalanceChanges`, because it doesn't create only the leave request, but also the sickness request